### PR TITLE
Replace deprecated JexlArithmetic.compare()

### PR DESCRIPTION
**Pull request by SethSmucker**
The previous version used a String instead of a JexlOperator for the operator parameter. The following classes have been updated to use JexlArithmetic.compare(Object left, Object right, JexlOperator operator):

- datawave.query.jexl.DatawaveArithmetic
- datawave.query.jexl.DefaultArithmetic
- datawave.query.jexl.HitListArithmetic

part of https://github.com/NationalSecurityAgency/datawave/issues/2443

### CODE REVIEW

**Review by avgAGB — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/HitListArithmetic.java [hunk 1]
```diff
@@ -221,7 +222,7 @@ public boolean lessThan(final Object left, final Object right) {
             for (final Object o : set) {
                 Object normalizedO = ValueTuple.getNormalizedValue(o);
                 Object normalizedRight = ValueTuple.getNormalizedValue(right);
-                if (compare(normalizedO, normalizedRight, LESS_THAN) < 0) {
+                if (compare(normalizedO, normalizedRight, JexlOperator.LT) < 0) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/jexl/HitListArithmetic.java]: I don't think the string constants on line 26 are needed anymore

File: warehouse/query-core/src/main/java/datawave/query/jexl/DefaultArithmetic.java [hunk 2]
```diff
@@ -178,7 +179,7 @@ public boolean lessThan(Object left, Object right) {
             Set<Object> set = (Set<Object>) left;
 
             for (Object o : set) {
-                if (super.compare(o, right, LESS_THAN) < 0) {
+                if (super.compare(o, right, JexlOperator.LT) < 0) {
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/jexl/DefaultArithmetic.java]: I don't think the string constants on line 16 are needed anymore

**Review by apmoriarty — DISMISSED** [review]

**Review by foster33 — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/HitListArithmetic.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/HitListArithmetic.java]: +1

File: warehouse/query-core/src/main/java/datawave/query/jexl/DefaultArithmetic.java (hunk 2 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/DefaultArithmetic.java]: +1

**Review by ivakegg — CHANGES_REQUESTED** [review]
Remove unused constants

File: warehouse/query-core/src/main/java/datawave/query/jexl/DefaultArithmetic.java (hunk 2 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/DefaultArithmetic.java]: Still not resolved

File: warehouse/query-core/src/main/java/datawave/query/jexl/HitListArithmetic.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/HitListArithmetic.java]: still not resolved

**Comment by ivakegg** [comment]
Why am I seeing changes to all of the docker files in the submodules in this PR.  Also the formatting is failing.

**Comment by SethSmucker** [comment]
> Why am I seeing changes to all of the docker files in the submodules in this PR. Also the formatting is failing.

I think that they're from changes merged from integration, but I could be wrong. I've been working on the sub-modules in their own directories, so that could be it too.

**Pull request closed without merging**